### PR TITLE
Enhance about

### PR DESCRIPTION
**Pull request by beroset**
This enhances the About page by adding many requested data items.  Anything that is not relevant or not available (e.g. the IMEI on watches without cellular network capability) is not displayed.  This, in combination with some other PRs addresses most of issue #33

### CODE REVIEW

File: src/qml/AboutPage.qml [hunk 1]
```diff
@@ -16,37 +17,78 @@
...
+            anchors.horizontalCenter: parent.horizontalCenter
+        }
+        Label {
+            id: releaseLabel
+            text: about.softwareVersion
+            opacity: 0.8
+            anchors.horizontalCenter: parent.horizontalCenter
+        }
+        Repeater {
+            model: [
```
Comment (FlorentRevest) [inline src/qml/AboutPage.qml:58]: Great :)

File: src/qml/AboutPage.qml [hunk 2]
```diff
@@ -16,37 +17,78 @@
...
+	    delegate: Column {
+		width: contentcolumn.width
+		anchors.horizontalCenter: contentcolumn.horizontalCenter
+		visible: modelData.text
+		Item {
+		    height: parent.width*0.05
+		    width: height
+		}
+		Label {
+		    id: myLabel
```
Comment (FlorentRevest) [inline src/qml/AboutPage.qml:81]: Did you intend to leave myLabel/myValue here? :) for some reason it's often used in tutorials but I find this naming pattern is never helpful

File: src/main.cpp [hunk 3]
```diff
@@ -28,12 +29,16 @@
...
+    utsname buf;
+    uname(&buf);
     QScopedPointer<QGuiApplication> app(AsteroidApp::application(argc, argv));
     QScopedPointer<QQuickView> view(AsteroidApp::createView());
     qmlRegisterType<VolumeControl>("org.asteroid.settings", 1, 0, "VolumeControl");
     qmlRegisterType<TiltToWake>("org.asteroid.settings", 1, 0, "TiltToWake");
     qmlRegisterType<TapToWake>("org.asteroid.settings", 1, 0, "TapToWake");
     view->setSource(QUrl("qrc:/qml/main.qml"));
+    view->rootContext()->setContextProperty("qtversion", QString(qVersion()));
+    view->rootContext()->setContextProperty("kernelversion", QString(buf.release));
```
Comment (FlorentRevest) [inline src/main.cpp:41]: In an ideal world these would be provided by a qml plugin but... that will do! :)

File: src/main.cpp [hunk 4]
```diff
@@ -28,12 +29,16 @@
...
 {
+    utsname buf;
+    uname(&buf);
     QScopedPointer<QGuiApplication> app(AsteroidApp::application(argc, argv));
     QScopedPointer<QQuickView> view(AsteroidApp::createView());
     qmlRegisterType<VolumeControl>("org.asteroid.settings", 1, 0, "VolumeControl");
     qmlRegisterType<TiltToWake>("org.asteroid.settings", 1, 0, "TiltToWake");
     qmlRegisterType<TapToWake>("org.asteroid.settings", 1, 0, "TapToWake");
     view->setSource(QUrl("qrc:/qml/main.qml"));
+    view->rootContext()->setContextProperty("qtversion", QString(qVersion()));
```
Comment (FlorentRevest) [inline src/main.cpp:40]: to keep the QML ids naming consistent, maybe qtVersion and kernelVersion ?

Comment (beroset) [reply src/main.cpp:40]: done!

File: src/qml/AboutPage.qml (hunk 1 above)
Comment (beroset) [reply src/qml/AboutPage.qml:58]: I was pleased with how this turned out, thanks!

File: src/qml/AboutPage.qml (hunk 2 above)
Comment (beroset) [reply src/qml/AboutPage.qml]: Good point.  I've renamed both to be `dataLabel` and `dataValue`.  It won't win an prizes for descriptive labels, but I couldn't really come up with something more descriptive in this context.

File: src/qml/AboutPage.qml [hunk 5]
```diff
@@ -16,37 +17,78 @@
...
+                { label: qsTr("Host name"), text: DeviceInfo.hostname },
+		{ label: qsTr("WLAN MAC"), text: about.wlanMacAddress },
+		{ label: qsTr("IMEI"), text: about.imei },
+                { label: qsTr("Serial number"), text: about.serial },
+		{ label: qsTr("Total disk space"), text: qsTr("%L1").arg(Math.round(about.totalDiskSpace() / 1e7)/100) + " GB" },
+		{ label: qsTr("Available disk space"), text: qsTr("%L1").arg(Math.round(about.availableDiskSpace() / 1e7)/100) 
+                            + " GB (" + (100.0 * about.availableDiskSpace() / about.totalDiskSpace()).toFixed(0) + "%)" },
+		{ label: qsTr("Display size"), text: Dims.w(100) + qsTr("W") + " x " + Dims.h(100) + qsTr("H") },
+		{ label: qsTr("Kernel version"), text: kernelversion },
+		{ label: qsTr("Qt version"), text: qtversion }
```
Comment (FlorentRevest) [inline src/qml/AboutPage.qml:70]: You fixed the setContextProperty but not the usages it seems ?

File: src/qml/AboutPage.qml [hunk 6]
```diff
@@ -16,37 +17,78 @@
...
+	    delegate: Column {
+		width: contentcolumn.width
+		anchors.horizontalCenter: contentcolumn.horizontalCenter
+		visible: modelData.text
+		Item {
+		    height: parent.width*0.05
+		    width: height
+		}
+		Label {
+		    id: dataLabel
```
Comment (FlorentRevest) [inline src/qml/AboutPage.qml:81]: How about no id at all ?

Earlier, the term "my" didn't mean anything but now "data" also doesn't seem to mean anything in the context of a Label. The best I could think of would be titleLabel and valueLabel but to keep things consistent then the model should also have title/value.

We don't really need ids for these anyway so maybe we can just skip the debate and not name them :)

Comment (beroset) [reply src/qml/AboutPage.qml:81]: Oh, I forgot we could just omit the names!

File: src/qml/AboutPage.qml (hunk 5 above)
Comment (beroset) [reply src/qml/AboutPage.qml]: Fixed, thanks!  I realize that my test failed to clear the QML cache.

**Comment by FlorentRevest** [comment]
It looks like the indentation of your code isn't consistent (some are tabs other are spaces) and I hadn't noticed so far because on the web github tabs are probably configured to look the same as in your editor but open in a different context this looks all messed up, could you fix the formatting before we merge this ?

**Comment by beroset** [comment]
Whitespace issue is now address and all tabs are converted to spaces.

**Comment by FlorentRevest** [comment]
Thank you beroset! :)

**Pull request merged**